### ISSUE TITLE
INT2 configuration mismatch on Atmega16

### ISSUE THREAD

**Issue by rubienr** [avrora-framework/avrora#186]
when enabling interrupts INT[0,2] in GIFR (0x5a) the simulation reports 

```
   0  0:00:00.00003075005   unpost interrupt: #4 (TIMER2 COMP)
```

but expected is:

```
   0  0:00:00.00003075005   unpost interrupt: #19 (INT2)
```

example snippet:

```
   0  0:00:00.00003075005  SRAM[GIFR.(INTF1 | INTF0 | INTF2 | - | - | - | - | -)] <- (0b11100000)
   0  0:00:00.00003075005  SRAM[5a] <- e0 @ 2a0
   0  0:00:00.00003075005  unpost interrupt: #4 (TIMER2 COMP)
   0  0:00:00.00003075005  unpost interrupt: #2 (INT0)
   0  0:00:00.00003075005  unpost interrupt: #3 (INT1)
```

class where INT2 is configured:
[ATMega16.java](https://github.com/avrora-framework/avrora/blob/master/avrora/src/main/java/edu/ucla/cs/compilers/avrora/avrora/sim/mcu/ATMega16.java)

**Comment by rubienr** [comment]
In [ATMega16.java](https://github.com/avrora-framework/avrora/blob/master/avrora/src/main/java/edu/ucla/cs/compilers/avrora/avrora/sim/mcu/ATMega16).installDevices()

In

```
 protected void installDevices() {
     // ...
     /**
      * mapping number corresponds to the (INTx_vect_num - 1)
      *  according to the iom16.h definition
      */
      int[] mapping = new int[]{-1, -1, -1, -1, -1, 4, 2, 3};
            // ...
      }
```

we expect the interrupt verctors to be defined as mentioned in **iom16.h**. It seems the vectors have been taken from an other header file where **INT2** is vector **4** but in **iom16.h** it is defined as follows:

```
/* External Interrupt Request 2 */
#define INT2_vect_num           18
#define INT2_vect               _VECTOR(18)
#define SIG_INTERRUPT2          _VECTOR(18)
```

Changeing the misleading documentation and mapping to

```
/**
 * mapping number corresponds to the (INTx_vect_num + 1)
 *  according to the iom16.h definition
 */
int[] mapping = new int[]{-1, -1, -1, -1, -1, 19, 2, 3};
```

should do the work.

A fast test of the fix results in the expected output:

```
    ...
0  0:00:00.00012425239  SRAM[GIFR.(INTF1 | INTF0 | INTF2 | - | - | - | - | -)] <- (0b11100000)
0  0:00:00.00012425239  SRAM[5a] <- e0 @ 3d8
0  0:00:00.00012425239  unpost interrupt: #19 (INT2)
0  0:00:00.00012425239  unpost interrupt: #2 (INT0)
0  0:00:00.00012425239  unpost interrupt: #3 (INT1)
    ...
0  0:00:00.00012537707  SRAM[GICR.(INT1 | INT0 | INT2 | - | - | - | IVSEL | IVCE)] <- (0b11100000)
0  0:00:00.00012537707  SRAM[5b] <- e0 @ 3ea
0  0:00:00.00012537707  enable interrupt: #19 (INT2)
0  0:00:00.00012537707  enable interrupt: #2 (INT0)
0  0:00:00.00012537707  enable interrupt: #3 (INT1)
   ...
```

**Comment by rubienr** [comment]
fixed in  b8c2aba in line 263
